src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -326,6 +374,31 @@ export const TradeConfirm = () => {
     )
   }, [handleBack, tradeStatus])
 
+  const donationOption: JSX.Element = useMemo(
+    () => (
+      <Stack spacing={4}>
+        <Row>
+          <HelperTooltip label={translate('trade.tooltip.donation')}>
+            <Row.Label>
+              <Checkbox isChecked={shouldDonate} onChange={handleDonationToggle}>
+                <Text translation='trade.donation' />
+              </Checkbox>
+            </Row.Label>
+          </HelperTooltip>
+          <Row.Value>{toFiat(donationAmountFiat ?? '0')}</Row.Value>
+        </Row>
+      </Stack>
+    ),
+    [donationAmountFiat, handleDonationToggle, shouldDonate, toFiat, translate],
+  )
+
+  const isTHORChainSwap = useMemo(
+    () => fees?.tradeFeeSource === SwapperName.Thorchain,
+    [fees?.tradeFeeSource],
+  )
+
+  const shouldShowDonationOption = useMemo(() => isTHORChainSwap, [isTHORChainSwap])
```

Comment:
nit pick - memoization has a cost and don't have to memoize primitives

Reply:
Fair - addressed in https://github.com/shapeshift/web/pull/4353/commits/886e7cd5c8727e4ab88c09019a750b8907790613